### PR TITLE
board: arm: Add support for Black Pill V1.2 F401CC

### PR DESCRIPTION
**Pull request by ric96**
Changes:

 - Copied over board files for blackpill_f401ce
 - Changed appropriate soc dtsi from stm32f401Xe.dtsi to stm32f401Xc.dtsi
 - Replaced any instance of F401CE to F401CC

In essense the blackpill_f401ce and blackpill_f401cc are the exact same boards,
the f401cc variant of the stm32 mcu has lesser sram and flash than the f401ce.

Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>

### CODE REVIEW

**Comment by ric96** [comment]
I did create this as a new board but let me know if all the three black pills, or at least the f401ce and f401cc, should be combined into one board  using overlays similar to some disco boards.

**Comment by erwango** [comment]
> I did create this as a new board but let me know if all the three black pills, or at least the f401ce and f401cc, should be combined into one board using overlays similar to some disco boards.

This would be nice, but I prefer to avoid to do this doing overlays as normally you can't change a soc (ie remove a .dtsi inclusion) using overlays (at least I think so).
This being said, maybe you can use the `boards/arm/stm32_min_dev` fashion, which at least group similar boards into the same folder).

**Comment by ric96** [comment]
> > I did create this as a new board but let me know if all the three black pills, or at least the f401ce and f401cc, should be combined into one board using overlays similar to some disco boards.
> 
> This would be nice, but I prefer to avoid to do this doing overlays as normally you can't change a soc (ie remove a .dtsi inclusion) using overlays (at least I think so). This being said, maybe you can use the `boards/arm/stm32_min_dev` fashion, which at least group similar boards into the same folder).

Hi. Sorry for the delay on this PR. I took a look at the `stm32-min-dev` bits and both black and blue variants share the same MCU as well, I was under the impression they didn't.
I think i'll fix the doc issue on this one for now. I think my main issue is that all the variants are called the "BlackPill".

**Comment by ric96** [comment]
@erwango hi, can you help a bit with this test failure.

File: boards/arm/blackpill_f401cc/blackpill_f401cc.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - gnuarmemb
+  - xtools
+supported:
+  - counter
+  - spi
+  - usb
+  - i2c
+  - uart
+  - pwm
+  - adc
```
Comment (erwango) [inline boards/arm/blackpill_f401cc/blackpill_f401cc.yaml:16]: Compilation issue comes from the fact that you mention adc is supported but the adc test does not support the board. 
Please update `tests/drivers/adc/adc_api/src/test_adc.c`

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/blackpill_f401cc/Kconfig.board [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020 Kalyan Sriram
```
[the remark this answers is not in the document]
Comment (ric96) [reply boards/arm/blackpill_f401cc/Kconfig.board:1]: Thanks for the clarification. I wasn't really sure about this so I kept the original author's tag in place. I have them changed now.

**Review by erwango — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**